orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -486,9 +456,12 @@ def _get_inference_graph(self, *inputs, **kwargs):
 
         # Setup dynamic axes for onnx model
         input_names, dynamic_axes, self._input_names_require_grad, _ = _parse_inputs_for_onnx_export(self._original_module_input_names, None, *inputs, **kwargs)
-        output_names, output_dynamic_axes, self._original_module_output_type = _parse_outputs_for_onnx_export(self._original_module, inputs, kwargs)
+        output_names, output_dynamic_axes, self._original_module_output_type, use_derived_module = _parse_outputs_for_onnx_export(self._original_module, inputs, kwargs)
         dynamic_axes.update(output_dynamic_axes)
 
+        if use_derived_module:
+            self._original_module = _module_output_transformation.DerivedModule(self._original_module)
```

Comment:
It is odd that the `_original_module` is not actually the original module, but an specialization of it